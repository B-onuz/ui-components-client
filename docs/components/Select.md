# Select (Component)

Componente de Select para formul√°rios.

## Como utilizar

```js
import Select from 'ui-components/lib/components/Select'
```

```jsx
<Select placeholder="Nome inteiro">
  <option value="valor1">Valor 1</option>
  <option value="valor2">Valor 2</option>
  <option value="valor3">Valor 3</option>
  <option value="valor4">Valor 4</option>
  <option value="valor5">Valor 5</option>
  <option value="valor6">Valor 6</option>
  <option value="valor7">Valor 7</option>
  <option value="valor8">Valor 8</option>
</Select>
```
