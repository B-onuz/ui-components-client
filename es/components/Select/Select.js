"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _propTypes2 = _interopRequireDefault(require("@styled-system/prop-types"));

var _variants = require("./variants");

var _color = _interopRequireDefault(require("color"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Select/Select.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  border: 1px solid #dededf;\n  appearance: none;\n  line-height: inherit;\n  border-radius: 4px;\n  line-height: 2.1em;\n  outline: none;\n  padding-right: 2em;\n  padding-left: 1em;\n  width: 100%;\n  box-sizing: border-box;\n  background: #fff;\n  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTc2cHgiIGhlaWdodD0iOTBweCIgdmlld0JveD0iMCAwIDE3NiA5MCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICAgIDx0aXRsZT5UcmlhbmdsZTwvdGl0bGU+CiAgICA8ZyBpZD0iV2VsY29tZSIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9IkFydGJvYXJkIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTYuMDAwMDAwLCAtNy4wMDAwMDApIiBmaWxsPSIjNEQ0RDREIj4KICAgICAgICAgICAgPHBhdGggZD0iTTEwNS40MTQyMTQsOC40MTQyMTM1NiBMMTkwLjU4NTc4Niw5My41ODU3ODY0IEMxOTEuMzY2ODM1LDk0LjM2NjgzNSAxOTEuMzY2ODM1LDk1LjYzMzE2NSAxOTAuNTg1Nzg2LDk2LjQxNDIxMzYgQzE5MC4yMTA3MTQsOTYuNzg5Mjg2MyAxODkuNzAyMDA2LDk3IDE4OS4xNzE1NzMsOTcgTDE4LjgyODQyNzEsOTcgQzE3LjcyMzg1NzYsOTcgMTYuODI4NDI3MSw5Ni4xMDQ1Njk1IDE2LjgyODQyNzEsOTUgQzE2LjgyODQyNzEsOTQuNDY5NTY3IDE3LjAzOTE0MDgsOTMuOTYwODU5MiAxNy40MTQyMTM2LDkzLjU4NTc4NjQgTDEwMi41ODU3ODYsOC40MTQyMTM1NiBDMTAzLjM2NjgzNSw3LjYzMzE2NDk4IDEwNC42MzMxNjUsNy42MzMxNjQ5OCAxMDUuNDE0MjE0LDguNDE0MjEzNTYgWiIgaWQ9IlRyaWFuZ2xlIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMDQuMDAwMDAwLCA1Mi4wMDAwMDApIHJvdGF0ZSgtMTgwLjAwMDAwMCkgdHJhbnNsYXRlKC0xMDQuMDAwMDAwLCAtNTIuMDAwMDAwKSAiPjwvcGF0aD4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg==');\n  background-repeat: no-repeat, repeat;\n  cursor: pointer;\n  background-position: right 0.7em top 50%, 0 0;\n  background-size: 0.65em auto, 100%;\n  text-overflow: ellipsis;\n  &::placeholder {\n    letter-spacing: 0px;\n    color: #b5b5b5;\n  }\n  &:focus {\n    box-shadow: 0 0 3px\n      ", ";\n    border-color: ", ";\n  }\n  ", "\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledSelect = _styledComponents["default"].select(_templateObject(), function (_ref) {
  var theme = _ref.theme,
      color = _ref.color,
      error = _ref.error;
  return error ? new _color["default"](theme.colors['red']).fade(0.5).string() : new _color["default"](theme.colors[color]).fade(0.5).string();
}, function (_ref2) {
  var theme = _ref2.theme,
      color = _ref2.color,
      error = _ref2.error;
  return error ? new _color["default"](theme.colors['red']).string() : theme.colors[color];
}, function (_ref3) {
  var error = _ref3.error,
      theme = _ref3.theme;
  return !!error && "\n      box-shadow: 0 0 3px " + new _color["default"](theme.colors['red']).fade(0.5).string() + ";\n      border: 1px solid " + theme.colors['red'] + ";\n    ";
}, _variants.size);

var Select = function Select(_ref4) {
  var rest = _extends({}, _ref4);

  return /*#__PURE__*/_react["default"].createElement(StyledSelect, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 45,
      columnNumber: 10
    }
  }));
};

Select.propTypes = process.env.NODE_ENV !== "production" ? _extends({}, _propTypes2["default"].size) : {};
Select.defaultProps = {
  size: 'medium',
  color: 'blue'
};
var _default = Select;
exports["default"] = _default;