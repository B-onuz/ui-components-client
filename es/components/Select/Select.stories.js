'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.SelectExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Select = _interopRequireDefault(require('./Select'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Select/Select.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Select',
  component: _Select['default'],
}
exports['default'] = _default

var SelectExample = function SelectExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      m: 20,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 11,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        width: [1 / 2],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 12,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Select['default'],
        {
          placeholder: 'Nome inteiro',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 13,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor4',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 14,
              columnNumber: 9,
            },
          },
          'Valor 1 exemplo teste com texto extenso no campo de op\xE7\xE3o '
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor2',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 15,
              columnNumber: 9,
            },
          },
          'Valor 2'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor3',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 16,
              columnNumber: 9,
            },
          },
          'Valor 3'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor1',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 17,
              columnNumber: 9,
            },
          },
          'Valor 4'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor5',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 18,
              columnNumber: 9,
            },
          },
          'Valor 5'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor6',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 19,
              columnNumber: 9,
            },
          },
          'Valor 6'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor7',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 20,
              columnNumber: 9,
            },
          },
          'Valor 7'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor8',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 21,
              columnNumber: 9,
            },
          },
          'Valor 8'
        )
      )
    )
  )
}

exports.SelectExample = SelectExample
