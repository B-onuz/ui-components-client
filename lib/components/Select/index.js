"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Select = _interopRequireDefault(require("./Select"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Select["default"];
exports["default"] = _default;
module.exports = exports.default;