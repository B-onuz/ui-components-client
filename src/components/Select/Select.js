import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import propTypes from '@styled-system/prop-types'
import { size } from './variants'
import Color from 'color'

const StyledSelect = styled.select`
  border: 1px solid #dededf;
  appearance: none;
  line-height: inherit;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 2em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  background: #fff;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTc2cHgiIGhlaWdodD0iOTBweCIgdmlld0JveD0iMCAwIDE3NiA5MCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICAgIDx0aXRsZT5UcmlhbmdsZTwvdGl0bGU+CiAgICA8ZyBpZD0iV2VsY29tZSIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9IkFydGJvYXJkIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTYuMDAwMDAwLCAtNy4wMDAwMDApIiBmaWxsPSIjNEQ0RDREIj4KICAgICAgICAgICAgPHBhdGggZD0iTTEwNS40MTQyMTQsOC40MTQyMTM1NiBMMTkwLjU4NTc4Niw5My41ODU3ODY0IEMxOTEuMzY2ODM1LDk0LjM2NjgzNSAxOTEuMzY2ODM1LDk1LjYzMzE2NSAxOTAuNTg1Nzg2LDk2LjQxNDIxMzYgQzE5MC4yMTA3MTQsOTYuNzg5Mjg2MyAxODkuNzAyMDA2LDk3IDE4OS4xNzE1NzMsOTcgTDE4LjgyODQyNzEsOTcgQzE3LjcyMzg1NzYsOTcgMTYuODI4NDI3MSw5Ni4xMDQ1Njk1IDE2LjgyODQyNzEsOTUgQzE2LjgyODQyNzEsOTQuNDY5NTY3IDE3LjAzOTE0MDgsOTMuOTYwODU5MiAxNy40MTQyMTM2LDkzLjU4NTc4NjQgTDEwMi41ODU3ODYsOC40MTQyMTM1NiBDMTAzLjM2NjgzNSw3LjYzMzE2NDk4IDEwNC42MzMxNjUsNy42MzMxNjQ5OCAxMDUuNDE0MjE0LDguNDE0MjEzNTYgWiIgaWQ9IlRyaWFuZ2xlIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMDQuMDAwMDAwLCA1Mi4wMDAwMDApIHJvdGF0ZSgtMTgwLjAwMDAwMCkgdHJhbnNsYXRlKC0xMDQuMDAwMDAwLCAtNTIuMDAwMDAwKSAiPjwvcGF0aD4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg==');
  background-repeat: no-repeat, repeat;
  cursor: pointer;
  background-position: right 0.7em top 50%, 0 0;
  background-size: 0.65em auto, 100%;
  text-overflow: ellipsis;
  &::placeholder {
    letter-spacing: 0px;
    color: #b5b5b5;
  }
  &:focus {
    box-shadow: 0 0 3px
      ${({ theme, color, error }) => (error ? new Color(theme.colors['red']).fade(0.5).string() : new Color(theme.colors[color]).fade(0.5).string())};
    border-color: ${({ theme, color, error }) => (error ? new Color(theme.colors['red']).string() : theme.colors[color])};
  }
  ${({ error, theme }) =>
    !!error &&
    `
      box-shadow: 0 0 3px ${new Color(theme.colors['red']).fade(0.5).string()};
      border: 1px solid ${theme.colors['red']};
    `}
  ${size}
`

const Select = ({ ...rest }) => {
  return <StyledSelect {...rest} />
}

Select.propTypes = {
  ...propTypes.size,
}

Select.defaultProps = {
  size: 'medium',
  color: 'blue',
}

export default Select
