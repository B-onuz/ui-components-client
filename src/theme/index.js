const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
