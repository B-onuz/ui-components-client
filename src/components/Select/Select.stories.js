import React from 'react'
import Select from './Select'
import { Flex, Box } from 'reflexbox'

export default {
  title: 'Select',
  component: Select,
}

export const SelectExample = () => (
  <Flex m={20}>
    <Box width={[1 / 2]}>
      <Select placeholder="Nome inteiro">
        <option value="valor4">Valor 1 exemplo teste com texto extenso no campo de opção </option>
        <option value="valor2">Valor 2</option>
        <option value="valor3">Valor 3</option>
        <option value="valor1">Valor 4</option>
        <option value="valor5">Valor 5</option>
        <option value="valor6">Valor 6</option>
        <option value="valor7">Valor 7</option>
        <option value="valor8">Valor 8</option>
      </Select>
    </Box>
  </Flex>
)
