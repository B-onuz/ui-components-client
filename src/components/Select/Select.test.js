import React from 'react'
import Select from './Select.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Select', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Select placeholder="Nome inteiro">
        <option value="valor1">Valor 1</option>
        <option value="valor2">Valor 2</option>
        <option value="valor3">Valor 3</option>
        <option value="valor4">Valor 4</option>
        <option value="valor5">Valor 5</option>
        <option value="valor6">Valor 6</option>
        <option value="valor7">Valor 7</option>
        <option value="valor8">Valor 8</option>
      </Select>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
