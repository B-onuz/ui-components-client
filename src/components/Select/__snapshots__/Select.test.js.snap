// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Select 1`] = `
.c0 {
  border: 1px solid #dededf;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  line-height: inherit;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 1em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  background: #fff;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTc2cHgiIGhlaWdodD0iOTBweCIgdmlld0JveD0iMCAwIDE3NiA5MCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICAgIDx0aXRsZT5UcmlhbmdsZTwvdGl0bGU+CiAgICA8ZyBpZD0iV2VsY29tZSIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9IkFydGJvYXJkIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTYuMDAwMDAwLCAtNy4wMDAwMDApIiBmaWxsPSIjNEQ0RDREIj4KICAgICAgICAgICAgPHBhdGggZD0iTTEwNS40MTQyMTQsOC40MTQyMTM1NiBMMTkwLjU4NTc4Niw5My41ODU3ODY0IEMxOTEuMzY2ODM1LDk0LjM2NjgzNSAxOTEuMzY2ODM1LDk1LjYzMzE2NSAxOTAuNTg1Nzg2LDk2LjQxNDIxMzYgQzE5MC4yMTA3MTQsOTYuNzg5Mjg2MyAxODkuNzAyMDA2LDk3IDE4OS4xNzE1NzMsOTcgTDE4LjgyODQyNzEsOTcgQzE3LjcyMzg1NzYsOTcgMTYuODI4NDI3MSw5Ni4xMDQ1Njk1IDE2LjgyODQyNzEsOTUgQzE2LjgyODQyNzEsOTQuNDY5NTY3IDE3LjAzOTE0MDgsOTMuOTYwODU5MiAxNy40MTQyMTM2LDkzLjU4NTc4NjQgTDEwMi41ODU3ODYsOC40MTQyMTM1NiBDMTAzLjM2NjgzNSw3LjYzMzE2NDk4IDEwNC42MzMxNjUsNy42MzMxNjQ5OCAxMDUuNDE0MjE0LDguNDE0MjEzNTYgWiIgaWQ9IlRyaWFuZ2xlIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMDQuMDAwMDAwLCA1Mi4wMDAwMDApIHJvdGF0ZSgtMTgwLjAwMDAwMCkgdHJhbnNsYXRlKC0xMDQuMDAwMDAwLCAtNTIuMDAwMDAwKSAiPjwvcGF0aD4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg==');
  background-repeat: no-repeat,repeat;
  cursor: pointer;
  background-position: right 0.7em top 50%,0 0;
  background-size: 0.65em auto,100%;
  font-size: 16px;
}

.c0::-webkit-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0::-moz-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0:-ms-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0::placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c0:focus {
  box-shadow: 0 0 3px rgba(33,133,208,0.5);
  border-color: #2185d0;
}

<select
  className="c0"
  color="blue"
  placeholder="Nome inteiro"
  size="medium"
>
  <option
    value="valor1"
  >
    Valor 1
  </option>
  <option
    value="valor2"
  >
    Valor 2
  </option>
  <option
    value="valor3"
  >
    Valor 3
  </option>
  <option
    value="valor4"
  >
    Valor 4
  </option>
  <option
    value="valor5"
  >
    Valor 5
  </option>
  <option
    value="valor6"
  >
    Valor 6
  </option>
  <option
    value="valor7"
  >
    Valor 7
  </option>
  <option
    value="valor8"
  >
    Valor 8
  </option>
</select>
`;
